File: app/src/main/java/com/riders/thelab/data/remote/api/GoogleAPIService.java
package com.riders.thelab.data.remote.api;

import com.riders.thelab.data.remote.dto.directions.Directions;


import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface GoogleAPIService {


    // Places

    // Directions
    @GET("directions/json")
    Single<Directions> getDirections(
            @Query("origin") String origin,
            @Query("destination") String destination,
            @Query("key") String key);
}

File: app/src/main/java/com/riders/thelab/data/remote/api/GoogleAPIService.kt
package com.riders.thelab.data.remote.api

import com.riders.thelab.data.remote.dto.directions.Directions
import io.reactivex.rxjava3.core.Single
import retrofit2.http.GET
import retrofit2.http.Query

interface GoogleAPIService {
    // Places

    // Places
    // Directions
    @GET("directions/json")
    fun getDirections(
            @Query("origin") origin: String,
            @Query("destination") destination: String,
            @Query("key") key: String?): Single<Directions>
}